pymatgen/entries/entry.py

```diff
@@ -0,0 +1,101 @@
+# coding: utf-8
+# Copyright (c) Pymatgen Development Team.
+# Distributed under the terms of the MIT License.
+
+"""
+This module implements the base Entry class, which is the basic entity
+that can be used to store calculated information. Other Entry classes 
+such as ComputedEntry and PDEntry inherit from this class.
+"""
+from pymatgen.core.composition import Composition
+from monty.json import MSONable
+from abc import ABCMeta, abstractmethod
+
+
+__author__ = "Shyue Ping Ong, Anubhav Jain, Ayush Gupta"
+__copyright__ = "Copyright 2020, The Materials Project"
+__version__ = "1.1"
+__maintainer__ = "Shyue Ping Ong"
+__email__ = "shyuep@gmail.com"
+__status__ = "Production"
+__date__ = "Mar 02, 2020"
+
+
+class Entry(MSONable, metaclass=ABCMeta):
+    """
+    A lightweight Entry object containing various types 
+    of data associated with a specific chemical composition.
```

Comment:
@ayushsgupta let's change this docstring to `A lightweight object containing the energy associated with a specific chemical composition. This base class is not intended to be instantiated directly. Note that classes which inherit from Entry must define a .energy property.`